File: src/Manager/LanguageManager.java
package Manager;

import java.util.Locale;
import java.util.ResourceBundle;

/**
 * @author Kevin
 */
public class LanguageManager {
    private static LanguageManager instance;
    
    private ResourceBundle bundle;
    
    private LanguageManager(){
        bundle = ResourceBundle.getBundle("Strings/Etiquetas_"+SettingsManager.getInstance().getLanguage());
    }
    
    public static LanguageManager getInstance(){
        if (instance == null){
            instance = new LanguageManager();
        }
        return instance;
    }
    
    public String get(String key){
        try {
            return bundle.getString(key);
        } catch (Exception e) {
            System.out.println(e.getMessage());
            return "Error";
        }
    }
    
    public void reloadLanguage(){
        bundle = ResourceBundle.getBundle("Strings/Etiquetas_"+SettingsManager.getInstance().getLanguage());
    }
}

File: src/main/kotlin/app/manager/LanguageManager.kt
package app.manager

import org.apache.logging.log4j.LogManager
import java.util.*

object LanguageManager {
    private val logger = LogManager.getLogger(LanguageManager::class.java)

    private const val BUNDLE_NAME = "locale/strings"
    private var bundle: ResourceBundle

    init {
        val locale = Locale(SettingsManager.getLanguage())
        bundle = ResourceBundle.getBundle(BUNDLE_NAME, locale)
    }

    fun get(key: String): String {
        try {
            return bundle.getString(key)
        } catch (e: Exception) {
            logger.warn("Property with key $key was not found.", e)
            return bundle.getString("key_not_found")
        }
    }

    fun reloadLanguage() {
        val locale = Locale(SettingsManager.getLanguage())
        bundle = ResourceBundle.getBundle(BUNDLE_NAME, locale)
    }
}